### PR TITLE
Change dependency to version 0.9.1 of OpenTelemetry API & SDK

### PR DESCRIPTION
**Pull request by zoercai**
A lot of breaking changes from 0.7.0 to 0.9.1: https://github.com/open-telemetry/opentelemetry-java/releases
The instrumentation library is at 0.8.0 still.

### CODE REVIEW

File: exporters/metrics/src/test/java/com/google/cloud/opentelemetry/metric/FakeData.java [hunk 1]
```diff
@@ -46,5 +40,10 @@
       .create("instrumentName", "0");
 
   static Point aLongPoint = LongPoint
-      .create(1599032114 * NANO_PER_SECOND, 1599031814 * NANO_PER_SECOND, Labels.empty(), 32L);
+      .create(1599032114 * NANO_PER_SECOND, 1599031814 * NANO_PER_SECOND,
+          Labels.of("label1", "value1", "label2", "False"), 32L);
+
+  static MetricData aMetricData = MetricData
```
Comment (nilebox) [inline exporters/metrics/src/test/java/com/google/cloud/opentelemetry/metric/FakeData.java:46]: nit: should it be `final static` (and same for other fields in this file)?

File: exporters/trace/src/main/java/com/google/cloud/opentelemetry/trace/TraceTranslator.java [hunk 2]
```diff
@@ -210,8 +209,8 @@ static Links toLinksProto(
   private static Link toLinkProto(io.opentelemetry.sdk.trace.data.SpanData.Link link) {
     checkNotNull(link);
     return Link.newBuilder()
-        .setTraceId(link.getContext().getTraceId().toLowerBase16())
-        .setSpanId(link.getContext().getSpanId().toLowerBase16())
+        .setTraceId(link.getContext().getTraceIdAsHexString())
```
Comment (nilebox) [inline exporters/trace/src/main/java/com/google/cloud/opentelemetry/trace/TraceTranslator.java:212]: We may need to confirm that the new trace/span ID format will be accepted by Cloud Trace.
Could you manually test that via an example against Cloud Trace?

**Review by nilebox — APPROVED** [review]

File: exporters/trace/src/main/java/com/google/cloud/opentelemetry/trace/TraceTranslator.java (hunk 2 above)
Comment (zoercai) [reply exporters/trace/src/main/java/com/google/cloud/opentelemetry/trace/TraceTranslator.java:212]: Yep ran `TraceExporterExample.java` with the current exporter-trace dependency and it was sent successfully to Cloud Trace:
image

**Comment by nilebox** [comment]
The CI test is failing with
```
com.google.api.gax.rpc.InvalidArgumentException: io.grpc.StatusRuntimeException: INVALID_ARGUMENT: displayName has max length of 128 bytes
	at com.google.api.gax.rpc.ApiExceptionFactory.createException(ApiExceptionFactory.java:49)
```

**Comment by zoercai** [comment]
@nilebox I've fixed the tests now, the displayName was just caused by the mock server needing the "opentelemetry/' prefix in the FakeData

File: exporters/trace/src/test/java/com/google/cloud/opentelemetry/trace/TraceTranslatorTest.java [hunk 3]
```diff
@@ -155,51 +155,10 @@ public Attributes getAttributes() {
...
-    Status spanStatus = TraceTranslator.toStatusProto(myStatus);
+    Status spanStatus = TraceTranslator.toStatusProto(OK);
 
     // The int representation is 0 for canonical code "OK".
     assertEquals(0,spanStatus.getCode());
-    assertEquals("Status description", spanStatus.getMessage());
-  }
-  
-  @Test
-  public void testToLinksProto(){
```
Comment (zoercai) [inline exporters/trace/src/test/java/com/google/cloud/opentelemetry/trace/TraceTranslatorTest.java:167]: From the Otel-java release:

> BREAKING CHANGE: Link and Event interfaces have been removed from the API

File: exporters/trace/src/test/java/com/google/cloud/opentelemetry/trace/TestSpanData.java [hunk 4]
```diff
@@ -106,23 +102,17 @@ public TestSpanData build() {
...
+        public abstract Builder setSpanId(String spanId);
 
-        /**
-         * Set the {@link TraceFlags} on this builder.
-         *
-         * @param traceFlags the trace flags.
-         * @return this.
-         */
-        public abstract Builder setTraceFlags(TraceFlags traceFlags);
+        public abstract Builder setSampled(boolean isSampled);
```
Comment (zoercai) [inline exporters/trace/src/test/java/com/google/cloud/opentelemetry/trace/TestSpanData.java:115]: Otel-java release:

> BREAKING CHANGE: SpanId, TraceId and TraceFlags are no longer used as instances, but only contain helper methods for managing conversion between Strings, bytes and other formats. SpanId and TraceId are now natively String-typed, and the TraceFlags is a single byte.

**Pull request merged**